diff -ur pavucontrol-5.0.orig/src/pavucontrol.cc pavucontrol-5.0/src/pavucontrol.cc
--- pavucontrol-5.0.orig/src/pavucontrol.cc	2021-08-15 19:06:48.000000000 +0200
+++ pavucontrol-5.0/src/pavucontrol.cc	2023-07-28 01:39:00.243215086 +0200
@@ -45,6 +45,8 @@
 #include "mainwindow.h"
 #include "pavuapplication.h"
 
+#include <gdk/gdkx.h>
+
 #include <unordered_map>
 #include <utility>
 #include <memory>
@@ -916,10 +918,12 @@
     tab_number = _tab_number;
     retry = _retry;
 
-    ca_context_set_driver(ca_gtk_context_get(), "pulse");
-
     mainWindow = MainWindow::create(maximize);
 
+    if (GDK_IS_X11_WINDOW(mainWindow->gobj())) {
+        ca_context_set_driver(ca_gtk_context_get(), "pulse");
+    }
+
     api = pa_glib_mainloop_get_api(m);
     g_assert(api);
 
diff -ur pavucontrol-5.0.orig/src/sinkwidget.cc pavucontrol-5.0/src/sinkwidget.cc
--- pavucontrol-5.0.orig/src/sinkwidget.cc	2021-08-15 19:06:48.000000000 +0200
+++ pavucontrol-5.0/src/sinkwidget.cc	2023-07-28 01:35:42.425595898 +0200
@@ -32,6 +32,9 @@
 
 #include "i18n.h"
 
+#include <gdk/gdkx.h>
+
+
 SinkWidget::SinkWidget(BaseObjectType* cobject, const Glib::RefPtr<Gtk::Builder>& x) :
     DeviceWidget(cobject, x) {
 #if HAVE_EXT_DEVICE_RESTORE_API
@@ -98,22 +101,24 @@
 
     pa_operation_unref(o);
 
-    ca_context_playing(ca_gtk_context_get(), 2, &playing);
-    if (playing)
-        return;
-
-    snprintf(dev, sizeof(dev), "%lu", (unsigned long) index);
-    ca_context_change_device(ca_gtk_context_get(), dev);
-
-    ca_gtk_play_for_widget(GTK_WIDGET(gobj()),
-                           2,
-                           CA_PROP_EVENT_DESCRIPTION, _("Volume Control Feedback Sound"),
-                           CA_PROP_EVENT_ID, "audio-volume-change",
-                           CA_PROP_CANBERRA_CACHE_CONTROL, "permanent",
-                           CA_PROP_CANBERRA_ENABLE, "1",
-                           NULL);
+    if (GDK_IS_X11_DISPLAY(this->get_display()->gobj())) {
+        ca_context_playing(ca_gtk_context_get(), 2, &playing);
+        if (playing)
+            return;
+
+        snprintf(dev, sizeof(dev), "%lu", (unsigned long) index);
+        ca_context_change_device(ca_gtk_context_get(), dev);
+
+        ca_gtk_play_for_widget(GTK_WIDGET(gobj()),
+                               2,
+                               CA_PROP_EVENT_DESCRIPTION, _("Volume Control Feedback Sound"),
+                               CA_PROP_EVENT_ID, "audio-volume-change",
+                               CA_PROP_CANBERRA_CACHE_CONTROL, "permanent",
+                               CA_PROP_CANBERRA_ENABLE, "1",
+                               NULL);
 
-    ca_context_change_device(ca_gtk_context_get(), NULL);
+        ca_context_change_device(ca_gtk_context_get(), NULL);
+    }
 }
 
 void SinkWidget::onMuteToggleButton() {
